app/models/user.rb

```diff
@@ -2,7 +2,8 @@ class User < ApplicationRecord
   has_many :comments
   has_many :posts, foreign_key: :author_id
   has_many :likes
-
+  validates :name, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
   # to use this method in rails c user = User.find(3); user.recent_posts
   def recent_posts
     posts.order(created_at: :desc).limit(3)
```

Comment:
- [ ] Great job so far @najibullahjafari There are still a few changes that need to be resolved and you'll be on your way to the next Milestone 😉 
- [ ] Just as the previous reviewer mentioned,  per the project requirements, you were asked to `Add unit specs for all of your models' methods and validations.` But currently, this `recent_posts` method in your `User Model` hasn't been tested. So kindly include tests for that method as well :wink: